### ISSUE TITLE
Proposal: releaseItem callback in constructor - Pooling support

### ISSUE THREAD

**Issue by ghost** [catamphetamine/virtual-scroller#10]
Is it possible to add `releaseItem` callback param in `DOMVirtualScroller ` constructor?
Callback should be called then every time library removes DOM Element from container.

It will help in **object pooling pattern** implementation.

**Comment by catamphetamine** [comment]
Hi. I didn’t understand what you meant. Can you describe in details, with
an example.

**Comment by ghost** [comment]
There is `renderItem `callback in `DOMVirtualScroller`, but there is no `releaseItem `callback.

Each time when `renderItem` is called, new DOM Element is created. And when you remove element: `this.container.removeChild(item)` code should call callback like `releaseItem` I proposed, so user of this library can implement **object pooling** for **DOM Element**:

https://en.wikipedia.org/wiki/Object_pool_pattern

This pattern will make Virtual-Scroller faster and GarbageCollector friendly.

**Comment by ghost** [comment]
Example:

```js
import VirtualScroller from 'virtual-scroller/dom'

const messages = [
  {
    username: ...,
    date: ...,
    text: ...
  },
  ...
];

const pool = [];

function createItemElement() {
  const element = document.createElement('article')

  const author = document.createElement('a')
  const time = document.createElement('time')
  const text = document.createElement('p')

  element.appendChild(author)
  element.appendChild(time)
  element.appendChild(text)

  return element
}

function renderItemElement(message) {
  const element = pool.pop() || createItemElement();

  const author = element.children[0];
  const time = element.children[1];
  const text = element.children[2];

  author.setAttribute('href', `/users/${message.username}`)
  author.textContent = `@${message.username}`
  
  time.setAttribute('datetime', message.date.toISOString())
  time.textContent = message.date.toString()

  text.textContent = message.text

  return element
}

function releaseItemElement(element) {
  pool.push(element);
}

const virtualScroller = new VirtualScroller(
  document.getElementById('messages'),
  messages,
  renderItemElement,
  releaseItemElement
)
```

**Comment by catamphetamine** [comment]
Ok, seems like this "pooling" thing could make some hypothetical sense.
https://github.com/ChrisAntaki/dom-pool#what-performance-gains-can-i-expect

See the related commit: will it resolve the issue?

**Comment by ghost** [comment]
Yes, it's great and works for sure. Thanks!

**Comment by catamphetamine** [comment]
Published `virtual-scroller@1.2.0`.
